deeplearning4j-core/src/main/java/org/deeplearning4j/datasets/iterator/MultipleEpochsIterator.java

```diff
@@ -211,7 +211,7 @@ public void setPreProcessor(org.nd4j.linalg.dataset.api.DataSetPreProcessor preP
      */
     @Override
     public boolean hasNext() {
-        if (newEpoch || (newEpoch && epochs == numEpochs)) {
+        if (newEpoch || epochs == numEpochs) {
             log.info("Epoch " + epochs + ", number of batches completed " + lastBatch);
```

Comment:
@nyghtowl I made these changes because they are logically wrong, and I wanted to open such discussions. 
`if (newEpoch || (newEpoch && epochs == numEpochs))` this logically has a problem, if `newEpoch` is `true` then all the if will be `true` and the if body will be executed, but if `newEpoch` is `false`, then it will go to check `(newEpoch && epochs == numEpochs)` which will always evaluate to `false`, because `newEpoch` is false!